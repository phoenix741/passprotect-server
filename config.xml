<?xml version='1.0' encoding='utf-8'?>
<widget id="org.shadoware.passprotect" version="1.4.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
    <name>Passprotect</name>
    <description>
      This application is used to store your password and credit card in a secure store
    </description>
    <author email="ulrich.vdh@shadoware.org" href="https://passprotect.shadoware.org">
      The website
    </author>
    <content src="index.html" />
    <plugin name="cordova-plugin-whitelist" spec="1" />
    <plugin name="cordova-plugin-splashscreen" spec="^5.0.1" />
    <access origin="*" />
    <access origin="https://stats.shadoware.org" />
    <allow-intent href="http://*/*" />
    <allow-intent href="https://*/*" />
    <allow-intent href="tel:*" />
    <allow-intent href="sms:*" />
    <allow-intent href="mailto:*" />
    <allow-intent href="geo:*" />
    <preference name="AutoHideSplashScreen" value="false" />
    <hook src="scripts/before_compile/build.js" type="before_compile" />
    <platform name="android">
        <allow-intent href="market:*" />
        <icon density="ldpi" src="res/android/icon/drawable-ldpi-icon.png" />
        <icon density="mdpi" src="res/android/icon/drawable-mdpi-icon.png" />
        <icon density="hdpi" src="res/android/icon/drawable-hdpi-icon.png" />
        <icon density="xhdpi" src="res/android/icon/drawable-xhdpi-icon.png" />
        <icon density="xxhdpi" src="res/android/icon/drawable-xxhdpi-icon.png" />
        <icon density="xxxhdpi" src="res/android/icon/drawable-xxxhdpi-icon.png" />
        <splash density="land-hdpi" src="res/android/splash/drawable-land-hdpi-screen.png" />
        <splash density="land-ldpi" src="res/android/splash/drawable-land-ldpi-screen.png" />
        <splash density="land-mdpi" src="res/android/splash/drawable-land-mdpi-screen.png" />
        <splash density="land-xhdpi" src="res/android/splash/drawable-land-xhdpi-screen.png" />
        <splash density="land-xxhdpi" src="res/android/splash/drawable-land-xxhdpi-screen.png" />
        <splash density="land-xxxhdpi" src="res/android/splash/drawable-land-xxxhdpi-screen.png" />
        <splash density="port-hdpi" src="res/android/splash/drawable-port-hdpi-screen.png" />
        <splash density="port-ldpi" src="res/android/splash/drawable-port-ldpi-screen.png" />
        <splash density="port-mdpi" src="res/android/splash/drawable-port-mdpi-screen.png" />
        <splash density="port-xhdpi" src="res/android/splash/drawable-port-xhdpi-screen.png" />
        <splash density="port-xxhdpi" src="res/android/splash/drawable-port-xxhdpi-screen.png" />
        <splash density="port-xxxhdpi" src="res/android/splash/drawable-port-xxxhdpi-screen.png" />
    </platform>
    <platform name="ios">
        <allow-intent href="itms:*" />
        <allow-intent href="itms-apps:*" />
        <icon height="180" src="res/ios/icon/icon-60@3x.png" width="180" />
        <icon height="60" src="res/ios/icon/icon-60.png" width="60" />
        <icon height="120" src="res/ios/icon/icon-60@2x.png" width="120" />
        <icon height="76" src="res/ios/icon/icon-76.png" width="76" />
        <icon height="152" src="res/ios/icon/icon-76@2x.png" width="152" />
        <icon height="40" src="res/ios/icon/icon-40.png" width="40" />
        <icon height="80" src="res/ios/icon/icon-40@2x.png" width="80" />
        <icon height="120" src="res/ios/icon/icon-40@3x.png" width="120" />
        <icon height="57" src="res/ios/icon/icon.png" width="57" />
        <icon height="114" src="res/ios/icon/icon@2x.png" width="114" />
        <icon height="72" src="res/ios/icon/icon-72.png" width="72" />
        <icon height="144" src="res/ios/icon/icon-72@2x.png" width="144" />
        <icon height="29" src="res/ios/icon/icon-small.png" width="29" />
        <icon height="58" src="res/ios/icon/icon-small@2x.png" width="58" />
        <icon height="87" src="res/ios/icon/icon-small@3x.png" width="87" />
        <icon height="50" src="res/ios/icon/icon-50.png" width="50" />
        <icon height="100" src="res/ios/icon/icon-50@2x.png" width="100" />
        <icon height="167" src="res/ios/icon/icon-83.5@2x.png" width="167" />
        <splash height="480" src="res/screen/ios/Default~iphone.png" width="320" />
        <splash height="960" src="res/screen/ios/Default@2x~iphone.png" width="640" />
        <splash height="1024" src="res/screen/ios/Default-Portrait~ipad.png" width="768" />
        <splash height="2048" src="res/screen/ios/Default-Portrait@2x~ipad.png" width="1536" />
        <splash height="2732" src="res/screen/ios/Default-Portrait@~ipadpro.png" width="2048" />
        <splash height="768" src="res/screen/ios/Default-Landscape~ipad.png" width="1024" />
        <splash height="1536" src="res/screen/ios/Default-Landscape@2x~ipad.png" width="2048" />
        <splash height="2048" src="res/screen/ios/Default-Landscape@~ipadpro.png" width="2732" />
        <splash height="1136" src="res/screen/ios/Default-568h@2x~iphone.png" width="640" />
        <splash height="1334" src="res/screen/ios/Default-667h.png" width="750" />
        <splash height="2208" src="res/screen/ios/Default-736h.png" width="1242" />
        <splash height="1242" src="res/screen/ios/Default-Landscape-736h.png" width="2208" />
    </platform>
    <engine name="android" spec="7.0.0" />
</widget>
