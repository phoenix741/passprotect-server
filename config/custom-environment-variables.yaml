config:
    mongodb:
        host: MONGODB_HOST
        user: MONGODB_USER
        password: MONGODB_PASSWORD
        options: MONGODB_OPTIONS

    jwt:
        secret: JWT_SECRET

    crypto:
        sessionKey: CRYPTO_SESSIONKEY
