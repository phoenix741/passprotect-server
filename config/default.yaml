config:
    mongodb:
        host: ~
        user: ~
        password: ~
        options: ~

    jwt:
        secret: 'dnLUMtULQsNmNbmGV3Lx8SxrxEtaxTc8aPdRh8YMemj515Faip7wQYueSaBFYm5r'

    crypto:
        cypherAlgorithm: 'AES-256-CTR'
        hashAlgorithm: 'sha256'
        sessionKey: 'xtipKI38GUCvE5cNGtTJxa1wQFvCicF5GDLTWyaBAb5RQqQ8rRBR1yVEq7Jg10cu'
