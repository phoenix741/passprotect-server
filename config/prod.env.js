const config = require('config')

module.exports = {
  NODE_ENV: '"production"',
  __PASSPROTECT_CONFIG__: JSON.stringify(config.get('config.client'))
}
