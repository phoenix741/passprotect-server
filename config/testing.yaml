config:
    mongodb:
        host: 'mongodb://localhost:27017/passprotect'
        user: ~
        password: ~
        options: ~
