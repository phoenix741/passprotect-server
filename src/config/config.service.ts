import * as Joi from '@hapi/joi';
import { MongooseOptionsFactory, MongooseModuleOptions } from '@nestjs/mongoose';
import { JwtOptionsFactory, JwtModuleOptions } from '@nestjs/jwt';

export interface EnvConfig {
  [key: string]: string;
}

export class ConfigService implements MongooseOptionsFactory, JwtOptionsFactory {
  private readonly envConfig: EnvConfig;

  constructor(environments: EnvConfig) {
    this.envConfig = this.validateInput(environments);
  }

  /**
   * Ensures all needed variables are set, and returns the validated JavaScript object
   * including the applied default values.
   */
  private validateInput(envConfig: EnvConfig): EnvConfig {
    const envVarsSchema: Joi.ObjectSchema = Joi.object({
      NODE_ENV: Joi.string()
        .valid(['development', 'production', 'test'])
        .default('development'),
      MONGODB_HOST: Joi.string().required(),
      MONGODB_DATABASE: Joi.string().optional(),
      JWT_SECRET: Joi.string().required(),
    });

    const { error, value: validatedEnvConfig } = Joi.validate(envConfig, envVarsSchema, { allowUnknown: true });
    if (error) {
      throw new Error(`Config validation error: ${error.message}`);
    }
    return validatedEnvConfig;
  }

  get mongodbHost(): string {
    return this.envConfig.MONGODB_HOST;
  }

  get mongodbDatabase(): string {
    return this.envConfig.MONGODB_DATABASE;
  }

  get jwtSecret(): string {
    return this.envConfig.JWT_SECRET;
  }

  createMongooseOptions(): MongooseModuleOptions {
    return {
      uri: this.mongodbHost + '/' + this.mongodbDatabase,
      useNewUrlParser: true,
    };
  }

  createJwtOptions(): JwtModuleOptions {
    return {
      secret: this.jwtSecret,
      signOptions: {
        expiresIn: 3600,
      },
    };
  }
}
