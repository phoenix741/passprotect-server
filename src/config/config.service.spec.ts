import { ConfigService } from './config.service';

describe('ConfigService', () => {
  describe('constructor', () => {
    let configService;

    it('missing variables', () => {
      expect(() => new ConfigService({})).toThrowError('');
    });

    it('all variables defined', () => {
      configService = new ConfigService({
        NODE_ENV: 'production',
        MONGODB_HOST: 'mongodb://localhost:27017',
        MONGODB_DATABASE: 'passprotect',
        JWT_SECRET: 'zaealsdfjklsklsdklEAZER13E31',
      });

      expect(configService.createMongooseOptions()).toMatchSnapshot(
        'mongooseOptions',
      );
      expect(configService.createJwtOptions()).toMatchSnapshot('jwtOptions');
    });
  });
});
