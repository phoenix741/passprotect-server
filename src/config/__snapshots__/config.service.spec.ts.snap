// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigService constructor all variables defined: jwtOptions 1`] = `
Object {
  "secret": "zaealsdfjklsklsdklEAZER13E31",
  "signOptions": Object {
    "expiresIn": 3600,
  },
}
`;

exports[`ConfigService constructor all variables defined: mongooseOptions 1`] = `
Object {
  "uri": "mongodb://localhost:27017/passprotect",
  "useNewUrlParser": true,
}
`;
